pkg/agent/flexvolume/controller.go

```diff
@@ -335,6 +335,17 @@ func (c *FlexvolumeController) GetClientAccessInfo(clusterName string, clientAcc
 	return nil
 }
 
+// GetKernelVersion returns the kernel version of the current node.
+func (c *FlexvolumeController) GetKernelVersion(_ *struct{} /* no inputs */, kernelVersion *string) error {
+	nodeName := os.Getenv(k8sutil.NodeNameEnvVar)
+	node, err := c.context.Clientset.Core().Nodes().Get(nodeName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to get kernel version from node information for node %s: %+v", nodeName, err)
+	}
+	*kernelVersion = node.Status.NodeInfo.KernelVersion
```

Comment:
I meant I fall back to treating it like an old kernel

Reply:
I meant I fall back to treating it like an old kernel

Reply:
@jbw976 im trying to avoid unnecessary checking of these value if possible since it will make the code messier with nested checks. 

This `nodeinfo` struct was added 2.5 years ago in https://github.com/kubernetes/kubernetes/pull/6211.

And node objects are part of the CoreAPI. Which means it shouldnt break or shouldnt give you uninitialize structs without giving you an error. Kinda like `Pod.Spec`.

If you still feel i should check for nil, i can add the checks.

Reply:
i was only asking you to look at the type definitions and see if any of them are pointers.  The answer is no:
https://github.com/kubernetes/api/blob/master/core/v1/types.go#L3878
https://github.com/kubernetes/api/blob/master/core/v1/types.go#L3683
https://github.com/kubernetes/api/blob/master/core/v1/types.go#L3634

checking for nil isn't necessary here.

Reply:
ahh ok.. yeah that is a good point.. I was mostly afraid of `kernelVersion` possible being empty in which case it would fall back to not using the `mds_namespace` option